cpp/src/io/utilities/config_utils.cpp

```diff
@@ -80,4 +87,36 @@ bool is_stable_enabled() { return is_all_enabled() or get_env_policy() == usage_
 
 }  // namespace nvcomp_integration
 
-}  // namespace cudf::io::detail
+inline std::mutex& host_mr_lock()
+{
+  static std::mutex map_lock;
+  return map_lock;
+}
+
+inline rmm::host_async_resource_ref default_pinned_mr()
+{
+  static rmm::mr::pinned_host_memory_resource default_mr{};
+  return default_mr;
+}
+
+inline auto& host_mr()
+{
+  static rmm::host_async_resource_ref host_mr = default_pinned_mr();
+  return host_mr;
+}
+
+}  // namespace detail
+
+void set_current_host_memory_resource(rmm::host_async_resource_ref mr)
```

Comment:
It would be great if this returned the old resource on the set. It would follow the pattern that was set for the device resources here: https://github.com/rapidsai/rmm/blob/f132d4b0daa976e1ec6cbcef24f5454fe510a394/include/rmm/mr/device/per_device_resource.hpp#L246

Reply:
not a blocker.

Reply:
Done.